frontend/src/metabase/core/components/EditableText/EditableText.tsx

```diff
@@ -41,6 +42,11 @@ const EditableText = forwardRef(function EditableText(
   const [submitValue, setSubmitValue] = useState(initialValue ?? "");
   const displayValue = inputValue ? inputValue : placeholder;
 
+  useEffect(() => {
+    setInputValue(initialValue ?? "");
+    setSubmitValue(initialValue ?? "");
+  }, [initialValue, setInputValue, setSubmitValue]);
+
```

Comment:
Done to support revisions on Dashboards. On Questions we effectively reload the whole query_builder component on revert, but I wasn't sure we would want to do that with Dashboards.

Reply:
That's actually a great idea... I've known about using keys in maps, but never thought to just add a key to an attribute to have it re-mount when needed. I've just added the whole description to the key. Do you think that's sufficient? Or should we run a checksum on the string to keep the length down?